### PR TITLE
Hit fox hub

### PR DESCRIPTION
**Pull request by nic-lan**
Official pull request :-)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,10 +15,20 @@ gem 'uglifier', '>= 1.3.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
 
+gem 'sidetiq'
+gem 'github_api'
+gem 'sinatra', :require => nil
```
Comment (DemitryT) [inline Gemfile:20]: Is sinatra needed for one of the gems?

File: app/features/hit_fox_feed.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require 'entities/github_feed'
```
Comment (DemitryT) [inline app/features/hit_fox_feed.rb:1]: this class should be in models/ or in the lib/ folder

File: app/features/hit_fox_feed.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+require 'jbuilder'
+
+class HitFoxFeed
+  def initialize(user,org,token)
+    @feed = GithubFeed.new(user,org, token)
```
Comment (DemitryT) [inline app/features/hit_fox_feed.rb:6]: extra space

File: app/workers/hit_fox_feed_worker.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+class HitFoxFeedWorker
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+
+  recurrence { minutely.second_of_minute(0, 5, 10, 15, 20, 25) }
+
+  def perform
+    feed = HitFoxFeed.new(ENV['GITHUB_USER'],'HitFox',ENV['GITHUB'])
+
+    File.write('app/views/hit_fox_feed/events.json', feed.events)
```
Comment (DemitryT) [inline app/workers/hit_fox_feed_worker.rb:12]: Why store this in a file? We should just cache the results. Also, the logic should be in `HitFoxFeed`

File: config/application.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 require File.expand_path('../boot', __FILE__)
 
 require 'rails/all'
+require 'dotenv' ; Dotenv.load '.env.development.env.local', ".env.#{Rails.env}"
```
Comment (DemitryT) [inline config/application.rb:4]: We can just use one `.env` file, and why is this needed actually?

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  # the seed, which is printed after each run.
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
```
Comment (DemitryT) [inline spec/spec_helper.rb:92]: eventually we can get rid of all of these comments

File: Gemfile (hunk 1 above)
Comment (nic-lan) [inline Gemfile:20]: sinatra is needed for the sidekiq monitoring

https://github.com/mperham/sidekiq/wiki/Monitoring

File: app/features/hit_fox_feed.rb (hunk 2 above)
Comment (nic-lan) [inline app/features/hit_fox_feed.rb:1]: it is actually in lib/entities :-)

i wasn't really sure about putting it there or in in the models directory.

File: app/features/hit_fox_feed.rb (hunk 3 above)
Comment (nic-lan) [inline app/features/hit_fox_feed.rb:6]: you mean the arguments for initialize i guess

File: app/workers/hit_fox_feed_worker.rb (hunk 4 above)
Comment (nic-lan) [inline app/workers/hit_fox_feed_worker.rb:12]: you're right !!!

about the cache: 
i can do that.
would you like a redis based one or you are thinking about something else ?

File: spec/spec_helper.rb (hunk 6 above)
Comment (nic-lan) [inline spec/spec_helper.rb:92]: you are right

File: app/features/hit_fox_feed.rb (hunk 2 above)
Comment (DemitryT) [inline app/features/hit_fox_feed.rb:1]: There's no need for the entities directory under lib, it can go directly there.

File: app/workers/hit_fox_feed_worker.rb (hunk 4 above)
Comment (DemitryT) [inline app/workers/hit_fox_feed_worker.rb:12]: Redis is fine since we already use it for sidekiq

File: lib/functions/symbolize_keys.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+def symbolize_keys(hash)
```
Comment (DemitryT) [inline lib/functions/symbolize_keys.rb:1]: This method should be in a module and under the `lib/modules` directory.

File: spec/features/hit_fox_hub_spec.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+require 'hit_fox_hub'
```
Comment (DemitryT) [inline spec/features/hit_fox_hub_spec.rb:2]: it doesn't get loaded automatically?
